controllers/oauth2.js

```diff
@@ -124,26 +134,42 @@ router.get('/consent', csrfProtection, auth.authChecker, (req, res, next) => {
 					value: scope,
 				})),
 			});
-		}).catch(next);
+		})
+		.catch(next);
 });
 
 router.post('/consent', auth.authChecker, (r, w) => acceptConsent(r, w, r.query.challenge, r.body.grantScopes, true));
 
-router.get('/username/:pseudonym', (req, res, next) => {
+router.get('/username/:pseudonym', async (req, res, next) => {
 	if (req.cookies.jwt) {
-		return api(req).get('/pseudonym', {
-			qs: {
-				pseudonym: req.params.pseudonym,
-			},
-		}).then((pseudonym) => {
-			let shortName;
-			let completeName;
-			const anonymousName = '???';
-			completeName = anonymousName;
-			shortName = completeName;
-			if (pseudonym.data.length) {
-				completeName = `${pseudonym.data[0].user.firstName} ${pseudonym.data[0].user.lastName}`;
-				shortName = `${pseudonym.data[0].user.firstName} ${pseudonym.data[0].user.lastName.charAt(0)}.`;
+		try {
+			let shortName = '???';
+			let completeName = '???';
+
+			if (Configuration.get('FEATURE_CTL_TOOLS_TAB_ENABLED')) {
+				const pseudonymResponse = await api(req, { version: 'v3' })
+					.get(`/pseudonyms/${req.params.pseudonym}`);
+
+				const userResponse = await api(req)
+					.get('/users', {
+						qs: { id: pseudonymResponse.userId },
+						$limit: 1,
+					});
```

Comment:
should we also add something like `if (userResponse.data?.length === 1) {` similar to `if (feathersPseudonymResponse.data.length) {` here?

Reply:
Good point. Changed that